### PR TITLE
Implement graylog - this is not finished. Part one of likely many.

### PR DESCRIPTION
**Pull request by ctheune**
@flyingcircusio/release-managers

Impact:

Hopefully nothing. This is a dark release of the "loghost" role for NixOS which is not in general use. Zagy will have to watch out for customer branch merges.

Changelog:

### CODE REVIEW

**Comment by ckauhaus** [comment]
@ctheune please rebase and resolve conflicts

**Comment by ctheune** [comment]
Done

File: nixos/lib/testing.nix [hunk 1]
```diff
@@ -131,8 +131,8 @@ rec {
       test = passMeta (runTests driver);
       report = passMeta (releaseTools.gcovReport { coverageRuns = [ test ]; });
 
-    in (if makeCoverageReport then report else test) // { 
-      inherit nodes driver test; 
+    in (if makeCoverageReport then report else test) // {
```
Comment (ckauhaus) [inline nixos/lib/testing.nix]: Useless whitespace change

File: nixos/modules/flyingcircus/packages/elasticsearch/plugins.nix [hunk 2]
```diff
@@ -0,0 +1,123 @@
+{ pkgs,  stdenv, fetchurl, fetchFromGitHub, unzip, elasticsearch }:
+
+with pkgs.lib;
+
+let
+  esPlugin = a@{
+    pluginName,
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/packages/elasticsearch/plugins.nix]: The Nix style guide advises to format long attrset parameter lists like this:
```
esPluing = a @
{ pluginName
, installPhase
...
}:
```

File: nixos/modules/flyingcircus/packages/fcmanage/default.nix [hunk 3]
```diff
@@ -11,6 +11,19 @@
 
 let
   py = python34Packages;
+  click = py.buildPythonPackage {
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/packages/fcmanage/default.nix]: Why here and not in python-packages.nix?

File: nixos/modules/flyingcircus/packages/fcmanage/fc/manage/graylog.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
+#! /run/current-system/sw/bin/env nix-shell
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/packages/fcmanage/fc/manage/graylog.py]: #!/usr/bin/env is available on all NixOS machines and should be used.

File: nixos/modules/flyingcircus/roles/loghost.nix [hunk 5]
```diff
@@ -0,0 +1,259 @@
+{ config, fclib, lib, pkgs, ... }: with lib;
```
Comment (ckauhaus) [inline nixos/modules/flyingcircus/roles/loghost.nix]: `with lib` is hard to spot while reading code -- better put it into a line of its own

**Pull request merged**